### DISCUSSION TITLE
Filter OpenAPI Generated Types

### DISCUSSION THREAD

**Discussion by penrodlol** [supabase/supabase#7747] [Questions] [answered]
I'm running: `npx openapi-typescript https://your-project.supabase.co/rest/v1/?apikey=your-anon-key --output types/supabase.ts`.

By default this command returns interfaces for paths, definitions, parameters, operations, and external. This is great but in the scope of my project, I really only care about the **definitions**. Other than doing some clever scripting to parse and remove content within the generated file, I'm unsure a clean way of doing this (if such a thing is possible)...

I'm also aware this may be just a limitation of openapi-typescript...

**Comment by steve-chavez** [comment, marked as answer]
Perhaps you could use [jq](https://stedolan.github.io/jq/) for this([discussed before](https://github.com/PostgREST/postgrest/issues/1421#issuecomment-560008636)). Like: 
```bash
curl https://your-project.supabase.co/rest/v1/?apikey=your-anon-key | jq '.definitions' > output.json

npx openapi-typescript output.json --output types/supabase.ts
```

**Reply by penrodlol** [reply]
Thanks, this worked!

I had to tweak jq to return swagger: "2.0" field as well. Without doing so, when running the openAPI spec through openapi-typescript, all the types would get wiped out...

Here is what mine ended up looking like:
`curl https://your-project.supabase.co/rest/v1/?apikey=your-anon-key | jq '{swagger,definitions}' > output.json`
